src/test/ui/methods/issues/issue-90315.rs

```diff
@@ -1,7 +1,54 @@
 fn main() {
-  let arr = &[0,1,2,3];
-  for _i in 0..arr.len().rev() { //~ERROR not an iterator
-     // The above error used to say “the method `rev` exists for type `usize`”.
-     // This regression test ensures it doesn't say that any more.
-  }
+    let arr = &[0, 1, 2, 3];
+    for _i in 0..arr.len().rev() { //~ERROR not an iterator
+        // The above error used to say “the method `rev` exists for type `usize`”.
+        // This regression test ensures it doesn't say that any more.
+    }
+
+    // Test
+    for i in 1..11.rev() {
+        //~^ ERROR not an iterator
+        // Suggest add the missing parentheses for range
+        println!("{i}");
+    }
+
+    let end: usize = 10;
+    for i in 1..end.rev() {
+        //~^ ERROR not an iterator
+        // Suggest add the missing parentheses for range
+        println!("{i}");
+    }
+
+    for i in 1..(end + 1).rev() {
+        //~^ ERROR not an iterator
+        // Suggest add the missing parentheses for range
+        println!("{i}");
+    }
+
+    if 1..(end + 1).is_empty() {
+        //~^ ERROR not an iterator
+        //~^^ ERROR mismatched types [E0308]
```

Comment:
```suggestion
        //~| ERROR mismatched types [E0308]
```

I think this should work. Change all occurrence of this to the one with `|`.